lfsapi/endpoint_finder.go

```diff
@@ -170,6 +170,9 @@ func (e *endpointGitFinder) NewEndpointFromCloneURL(rawurl string) Endpoint {
 
 func (e *endpointGitFinder) NewEndpoint(rawurl string) Endpoint {
 	rawurl = e.ReplaceUrlAlias(rawurl)
+	if strings.HasPrefix(rawurl, "/") {
+		return endpointFromLocalPath(rawurl)
```

Comment:
@sprohaska thanks for your clarification in https://github.com/git-lfs/git-lfs/pull/2912#issuecomment-377924331, it was very helpful for my understanding of this change. I have one remaining concern pertaining to this line. Since Git LFS does not support natively pushing or pulling without a server, I would like the user to be warned of this.

What do you think about warning when there are no custom transfer agents enabled when we're in this conditional? 

Something like:

```ShellSession
$ git lfs push
warning: Git LFS does not support pushing to local files without
warning: a custom transfer agent. Please see <url> for more.
# ...
```

Reply:
A warning makes sense. But I'm not sure whether this is the right place to print it. Maybe the warning could be printed by the code that decides whether to use a standalone agent or contact a server. If the decision there is to contact a server for a `file://` URL, a warning would be printed instead of trying to connect to it. I'll look into it in more detail later.

Reply:
I've added a commit that adds a hint about file:// and standalone transfer agent.